packages/wallet-protocols/src/protocols/create-and-direct-fund/protocol.ts

```diff
@@ -0,0 +1,122 @@
+import { Machine, MachineConfig } from 'xstate';
+
+import { AllocationAssetOutcome } from '@statechannels/nitro-protocol';
+
+import { MachineFactory } from '../../machine-utils';
+import { Store, success } from '../..';
+
+import { Participant } from '../../store';
+
+import { AdvanceChannel, DirectFunding } from '..';
+
+const PROTOCOL = 'create-and-direct-fund';
+
+export enum Indices {
+  Left = 0,
+  Right = 0,
+}
+
+export type Init = {
+  participants: Participant[];
+  allocations: AllocationAssetOutcome[];
+  appDefinition: string;
+  appData: string;
+  channelId: string;
+  challengeDuration: number;
+  index: Indices;
+};
+
+export type FirstStateConstructed = Init & { type: 'FIRST_STATE_CONSTRUCTED' };
+
+export const advanceChannelArgs = (i: 1 | 3) => ({ channelId }: Init): AdvanceChannel.Init => ({
+  channelId,
+  targetTurnNum: i,
+});
+
+const constructFirstState = {
+  invoke: {
+    id: 'constructFirstState',
+    src: 'constructFirstState',
+    onDone: 'preFundSetup',
+  },
+};
+
+const preFundSetup = {
+  invoke: {
+    id: 'preFundSetup',
+    src: 'advanceChannel',
+    data: advanceChannelArgs(1),
+    onDone: 'directFunding',
+  },
+  on: { CHANNEL_CLOSED: 'abort' },
+};
+
+// FIXME: Abort should not be success
+const abort = success;
+
+const directFunding = {
+  invoke: {
+    src: 'directFunding',
+    data: ({ allocations, channelId }: FirstStateConstructed): DirectFunding.Init => {
+      return {
+        channelId,
+        // TODO: Error handling on non-0 case ... also multi-asset
+        minimalAllocation: allocations[0].allocationItems,
```

Comment:
There is a `getEthAllocation` function that could be used. Using it would help keep track of where updates are needed for multi-asset protocols.